jpo-ode-common/src/main/java/us/dot/its/jpo/ode/util/JsonUtils.java

```diff
@@ -45,32 +49,45 @@ public JsonUtilsException(String string, Exception e) {
 
    }
 
-   private static Gson gsonCompact;
-   private static Gson gsonVerbose;
    private static ObjectMapper mapper;
+   private static ObjectMapper mapper_noNulls;
    private static Logger logger;
 
    private JsonUtils() {
       logger = LoggerFactory.getLogger(JsonUtils.class);
    }
 
    static {
-      gsonCompact = new GsonBuilder().create();
-      gsonVerbose = new GsonBuilder().serializeNulls().create();
       mapper = new ObjectMapper();
+      mapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+      mapper.coercionConfigFor(LogicalType.Enum)
+            .setCoercion(CoercionInputShape.EmptyString, CoercionAction.AsNull);
+
+      mapper_noNulls = new ObjectMapper();
+      mapper_noNulls.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+      mapper_noNulls.setSerializationInclusion(Include.NON_NULL);
    }
 
    public static String toJson(Object o, boolean verbose) {
-
       // convert java object to JSON format,
       // and returned as JSON formatted string
-      return verbose ? gsonVerbose.toJson(o) : gsonCompact.toJson(o);
+      try {
+         return verbose ? mapper.writeValueAsString(o) : mapper_noNulls.writeValueAsString(o);
```

Comment:
Not sure why we need this update. It does the same thing as the existing code.

Reply:
This swaps over to using the Jackson library. There was an odd mixture using both the Gson and Jackson libraries, and they did not always mix. When I got the unit tests to run, large numbers of them were failing and many were due to this mismatch of serializations. You are correct that the new code performs the same as previous in this instance and that was intentional - this update ensures compatibility and chooses a single serializer (Jackson) over the current duality.

Reply:
Thanks!